pkg/lifecycle: Add SetPredictionHeaders method

krisis opened the pull request:
## Description

This method is used to add expected expiration and transition time for an object
in Get/Head Object response headers.

Also fixed bugs in lifecycle.PredictTransitionTime and
getLifecycleTransitionTier in handling current and non-current versions.

## How to test this PR?


## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [x] Unit tests added/updated

krisis committed:
pkg/lifecycle: Add SetPredictionHeaders method

This method is used to add expected expiration and transition time for an object
in Get/Head Object response headers.

Also fixed bugs in lifecycle.PredictTransitionTime and
getLifecycleTransitionTier in handling current and non-current versions.

poornas commented on internal/bucket/lifecycle/lifecycle.go:
```diff
@@ -445,18 +448,42 @@ func (lc Lifecycle) PredictTransitionTime(obj ObjectOpts) (string, time.Time) {
-		case !rule.Transition.IsDaysNull():
+
+		case obj.IsLatest && !rule.Transition.IsDaysNull():
+			expectedTransition := ExpectedExpiryTime(obj.ModTime, int(rule.Expiration.Days))
```
```suggestion
			expectedTransition := ExpectedExpiryTime(obj.ModTime, int(rule.Transition.Days))
```

poornas commented on internal/bucket/lifecycle/lifecycle.go:
```diff
@@ -445,18 +448,42 @@ func (lc Lifecycle) PredictTransitionTime(obj ObjectOpts) (string, time.Time) {
+			}
+
+		case !obj.IsLatest && !rule.NoncurrentVersionTransition.IsDaysNull():
 			expectedTransition := ExpectedExpiryTime(obj.ModTime, int(rule.Expiration.Days))
```
```suggestion
			expectedTransition := ExpectedExpiryTime(obj.ModTime, int(rule.NoncurrentVersionTransition.Days))
```

krisis committed:
Address review comments

Fixed the incorrect transition time selection by providing NextDue method on
transition config types.
